src/test/java/org/javarosa/core/model/test/CyclicReferenceTests.java

```diff
@@ -0,0 +1,34 @@
+package org.javarosa.core.model.test;
+
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.xform.parse.XFormParseException;
+import org.junit.Test;
+
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for cyclic references
+ *
+ * @author wpride
+ */
+
+public class CyclicReferenceTests {
+    /**
+     * Test that XPath cyclic reference that references parent throws usable error
+     */
+    @Test
+    public void testCyclicReferenceWithGroup() {
+        try {
+            new FormParseInit("/xform_tests/group_cyclic_reference.xml");
+        } catch (XFormParseException e) {
+            String detailMessage = e.getMessage();
+            // Assert that we're using the shortest cycle algorithm
+            assert detailMessage.contains("Shortest Cycle");
+            // There should only be three newlines since only the three core cyclic references were included
+            int newlineCount = detailMessage.length() - detailMessage.replace("\n", "").length();
+            assert newlineCount == 3;
```

Comment:
I've never seen us use this syntax in tests before; does it result in the same sort of test failure and error message as `Assert.assertEquals(expected, actual)`?

Reply:
Hmm I hadn't thought about - I think `assert` is Java keyword rather than `Assert.assert*` which is JUnit specific. Not sure if they're behaviorally different, but changing to JUnit for consistency.